### PR TITLE
Pytest skip cleanup

### PR DESCRIPTION
**Pull request by nulano**
For #4193 

Changes proposed in this pull request:

 * Replace some `skipif`s with `skip_unless_feature`.
 * Rename `test_image_resample.py` test cases, as they weren't being picked up by pytest.
 * Use [`xfail`](https://docs.pytest.org/en/latest/skipping.html#xfail-mark-test-functions-as-expected-to-fail) instead of `skip_known_bad_test` and where appropriate for known failing tests. This lets pytest run the test anyway and report either XFAIL or XPASS without affecting the overall status. Add `-ra` pytest parameter to output xfail summary ([example here](https://github.com/nulano/Pillow/runs/770407250?check_suite_focus=true#step:22:1843)). Could help catch cases such as https://github.com/python-pillow/Pillow/pull/4696#issuecomment-643763461.

### CODE REVIEW

**Comment by nulano** [comment]
I can't figure out the Lint failure; `isort` is ok when run on my system.

File: Tests/test_file_palm.py [hunk 1]
```diff
@@ -2,15 +2,10 @@
 import subprocess
 
 import pytest
+
 from PIL import Image
```
Comment (radarhere) [inline Tests/test_file_palm.py:6]:
```suggestion
import pytest
from PIL import Image
```
I find that removing the blank line fixes isort

**Comment by hugovk** [comment]
Noting:

`pytest -rs` = show extra test summary: skipped
`pytest -ra` = show extra test summary: all except passed

```console
$ pytest --help
...
  -r chars              show extra test summary info as specified by chars: (f)ailed, (E)rror, (s)kipped, (x)failed,
                        (X)passed, (p)assed, (P)assed with output, (a)ll except passed (p/P), or (A)ll. (w)arnings are
                        enabled by default (see --disable-warnings), 'N' can be used to reset the list. (default: 'fE').
...
```

File: Tests/test_imagegrab.py [hunk 2]
```diff
@@ -4,8 +4,7 @@
 
 import pytest
 from PIL import Image, ImageGrab
-
-from .helper import assert_image, assert_image_equal_tofile
+from .helper import assert_image, assert_image_equal_tofile, skip_unless_feature
```
Comment (hugovk) [inline Tests/test_imagegrab.py:7]: isort lint fix:
```suggestion

from .helper import assert_image, assert_image_equal_tofile, skip_unless_feature
```

**Pull request merged**